src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/LspEditorFeatureDetector.cs

```diff
@@ -33,9 +42,19 @@ internal sealed class LspEditorFeatureDetector : ILspEditorFeatureDetector, IDis
         _jtf = joinableTaskContext.Factory;
         _disposeTokenSource = new();
 
+        _projectCapabilityResolver = projectCapabilityResolver;
+        _vsUIShellOpenDocument = new Lazy<IVsUIShellOpenDocument>(() =>
+        {
+            var shellOpenDocument = (IVsUIShellOpenDocument)ServiceProvider.GlobalProvider.GetService(typeof(SVsUIShellOpenDocument));
```

Comment:
Presumably this is fine to get this service off the UI thread?

Reply:
Incoming `JTF.Run` regression. <homer.hide>. It is only fine if the provider for this service is synchronously loaded instead of async.

Reply:
Perhaps moving this initialization into the existing AsyncLazy would be better?

Reply:
This is exactly what the code was doing before - was there an issue in the old code?

Reply:
Hmm, I guess maybe - 
![image](https://github.com/dotnet/razor/assets/1863656/4aeca4c1-ebdd-452f-a066-114b844cddca)

Reply:
Seems a bit messy to call from the existing async lazy as the tasks are somewhat orthogonal, with existing method calculating specific value. Just as existing code stipulates, first call always comes in on the main thread ```// This method is first called by out IFilePathToContentTypeProvider.TryGetContentTypeForFilePath(...) implementations.```, so I left it as is but added ThreadHelper.ThrowIfNotOnUIThread() which was the threading analyzer suggested fix.

I think it's good for this PR because that's *exactly* what existing code was already doing. If you folks disagree, we can clean up more in a follow-up PR.

Reply:
I _think_ given that this is a Lazy, and therefore not asyncronous in any way, the `ThrowIfNotOnUIThread` should do it.  A val build should confirm if we're good.

Reply:
Got val build running

Reply:
no issues with RPS or speedometer in val build, all passed.

Reply:
thanks for checking. infraswat appreciates you